version:  '3.8'

services:
  mysql_test:
    image: mysql:8.0
    restart: on-failure
    container_name: ${MYSQL_CONTAINER}
    environment:
      MYSQL_ROOT_PASSWORD: ${MYSQL_PASSWORD}
      MYSQL_DATABASE: ${MYSQL_DATABASE}
      MYSQL_PASSWORD: ${MYSQL_PASSWORD}
      MYSQL_USER: ${MYSQL_USER}
    ports:
      - '${MYSQL_PORT}:3306'
    expose:
      - '3306'
    volumes:
      - database-volume-test:/var/lib/mysql

  php_test:
    build:
      context: .
      dockerfile: docker/php/Dockerfile
    volumes:
      - './apps/symfony/:/usr/src/app'
    restart: on-failure
    env_file:
      - .env.test
    environment:
      DATABASE_URL: 'mysql://${MYSQL_USER}:${MYSQL_PASSWORD}@${MYSQL_CONTAINER}:3306/${MYSQL_DATABASE}?serverVersion=8.0.24'
    user: ${LOCAL_USER}
    depends_on:
      - mailer_test
      - mysql_test

  mailer_test:
    image: mailhog/mailhog:latest
    container_name: mailhog_test
    restart: on-failure
    ports:
      - "1026:1025"
      - "8026:8025"

volumes:
  database-volume-test:
